Baitap1/script.js

```diff
@@ -0,0 +1,268 @@
+    document.getElementById("formR").reset();
+    location.reload();
+}
+
+// alert refuse
+function refuse(inputName, labelName) {
+    labelName.innerHTML = "&#8861;";
+    labelName.style.color = "red";
+    inputName.style.border = "1px solid red";
+}
+
+// alert agree
+function agree(inputName, labelName) {
+    labelName.innerHTML = "&checkmark;";
+    labelName.style.color = "lightgreen";
+    inputName.style.border = "1px solid lightgreen";
+}
+
+//function load image
+function convertImg(image) {
+    const imageUpload = document.getElementById(image);
+    reader.addEventListener("load", function () {
+        // convert image file to base64 string
+        imageUpload.src = reader.result;
+    }, false);
+}
+function imageLoad() {
+    const titleImg = document.getElementsByClassName("centered")[0];
+    const image = convertImg("imageUpload");
+    var formData = new FormData();
+    var file = document.getElementById("imgLoad").files[0];
+    formData.append("Filedata", file);
+    var t = file.type.split('/').pop().toLowerCase();
+
+    if (t != "jpeg" && t != "jpg" && t != "png" && t != "bmp" && t != "gif") {
+        alert('Please select a valid image file');
+        document.getElementById("imgLoad").value = '';
+        return false;
+    } else {
+        reader.readAsDataURL(file);
+        titleImg.style.display = "none";
+        return true;
+    }
+}
+
+//check email
+function checkEmail() {
+    const regexEmail = /^[^<>()[\]\\,;:\%\_\.\*\{\}\[\]\|\/\+\=\?\'#^\s@\"$&!@]+@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z0-9]+\.)+[a-zA-Z]{2,3}))$/;
+    let inputEmail = document.getElementById("inputEmail");
+    let emailAlert = document.getElementById("emailAlert");
+    let checkRegex = regexEmail.test(inputEmail.value);
+
+    if (checkRegex) {
+        agree(inputEmail, emailAlert);
+        return true;
+    } else {
+        refuse(inputEmail, emailAlert);
+        return false;
+    }
+}
+
+//function check input name
+function processInputName(idName, idInput) {
+    const lowercaseRegex = /[a-z]|á|à|ạ|ã|ả|ă|ắ|ằ|ẳ|ặ|ẵ|â|ấ|ầ|ẫ|ẩ|ậ|é|è|ẽ|ẻ|ẹ|ê|ế|ề|ể|ễ|ệ|ì|í|ĩ|ị|ỉ|ò|ó|ỏ|õ|ọ|ô|ố|ồ|ộ|ổ|ỗ|ơ|ớ|ờ|ở|ợ|ỡ|ú|ù|ủ|ụ|ũ|ư|ứ|ừ|ữ|ử|ự|ý|ỳ|ỹ|ỵ|ỷ|đ/g;
+    const uppercaseRegex = /[A-Z]|Á|À|Ạ|Ã|Ả|Ă|Ắ|Ằ|Ẳ|Ặ|Ẵ|Â|Ấ|Ầ|Ẫ|Ẩ|Ậ|É|È|Ẽ|Ẻ|Ẹ|Ê|Ế|Ề|Ể|Ễ|Ệ|Ì|Í|Ĩ|Ị|Ỉ|Ò|Ó|Ỏ|Õ|Ọ|Ô|Ố|Ồ|Ộ|Ổ|Ỗ|Ơ|Ớ|Ờ|Ở|Ợ|Ỡ|Ú|Ù|Ủ|Ụ|Ũ|Ư|Ứ|Ừ|Ữ|Ử|Ự|Ý|Ỳ|Ỹ|Ỵ|Ỷ|Đ/g;
+    const findNumber = /[0-9]/g;
+    const specialCharacter = /\~|\`|\!|\@|\#|\$|\%|\^|\&|\*|\(|\)|\_|\-|\+|\=|\{|\[|\}|\]|\||\\|\:|\;|\"|\'|\<|\,|\>|\.|\?|\//;
+    let inputName = document.getElementById(idInput);
+    let alertName = document.getElementById(idName);
+    var text = inputName.value.trim();
+    var checkValue;
+
+    if (inputName.value == "") {
+        inputName.style.border = "1px solid red";
+        alertName.innerHTML = "&#8861;";
+        alertName.style.color = "red";
+        return false;
+    } else {
+        for (let i = 0; i < text.length; i++) {
```

Comment:
e dùng match hoặc test để validate được rồi, dùng for check từng kí tự cho rườm rà e